frog/imports/ui/Repository.jsx

```diff
@@ -1,9 +1,230 @@
+    event.preventDefault();
+  }
+
+  renderListActivities() {
+
+    var typesLowerCase = this.state.type.map(type => type.toLowerCase())
+
+    return (
+      this.props.activities ?
+        this.props.activities
+          //Filters activities with a name containing the text from the nameFilter input.
+          .filter((activity) => activity.name.indexOf(this.state.nameFilterText) != -1)
+          //Filters activities with respect to the list of filters
+          .filter((activity) => this.state.plane.indexOf(activity.plane) == -1)
+          .filter((activity) => typesLowerCase.indexOf(activity.type) == -1)
+          .map ((activity) => (
+            <Activity
+              key={activity._id}
+              id={activity._id}
+              type={activity.type}
+              name={activity.name}
+              plane={activity.plane}
+              object={activity.object} />
+          )) : <li>No activity</li>
+    );
+  }
+
+  render() {
+
+    //To add a new filter for the planes or types, just add the corresponding
+    //checkbox by using a createFilterBox and the rest is taken care of by
+    //the code.
+    return (
+        <div>
+          <h2>Activities:</h2>
+
+          <form className="input-filter-name"
+            onInput={this.handleFilterName.bind(this)}
+            onSubmit={this.handleFilterNameSubmit.bind(this)}>
+
+            <input
+              type="text"
+              ref ="nameFilter"
+              placeholder="Filter by name"
+            /><br/><br/>
+
+            <div className="filters">
+            Show only plane(s):
+              {this.createFilterBoxPlane(1)}
+              {this.createFilterBoxPlane(2)}
+              {this.createFilterBoxPlane(3)}
+              <br/>
+            Show only the following type(s) of activities:
+              {this.createFilterBoxType("Lecture")}
+              {this.createFilterBoxType("Quizz")}
+              {this.createFilterBoxType("Video")}
+            </div>
+
+          </form>
+          <ul>
+            {this.renderListActivities()}
+          </ul>
+        </div>
+      );
+  }
+
+
+}
+
+class Activity extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isClicked: false,
+    }
+  }
+
+  activityHandler(event) {
+    event.preventDefault();
+    this.setState({
+      isClicked: !this.state.isClicked,
+    });
+  }
+
+  renderObjectProperties(object, key) {
+        var properties = []
+        var i = 0;
+        for(var prop in object) {
+          if(object.hasOwnProperty(prop)) {
+            //Iterates over properties of the object (as we don't know what they are)
+            //These special keys are created so that they are unique for each element of the array
+            properties.push(React.DOM.br({key: key + "-br" + i}));
+            properties.push(React.DOM.a({key: key + "-a" + i}, prop + ": "+ object[prop]));
+            ++i;
+          }
+        }
+
+    return properties;
+  }
```

Comment:
Hey, I don't really like the use of `React.DOM.a` and `React.DOM.br` instead of the xml tags. What do you think of the code below ?

```
  renderObjectProperties(object) {
    class Property extends Component { 
      render () {
        return(
          <div>
            <br/>
            <a>{this.props.data.prop + ": "+ this.props.data.value}</a>
          </div>
        )
      }
    }

    return(Object.keys(object).map((prop)=>(
      <Property key={prop} data={{prop:prop, value:object[prop]}}/>
    )))
  }
```